### ISSUE TITLE
Emit New Event with Message Hash and ID on DMP Message Sent

### ISSUE THREAD

**Issue by XY-Wang** [paritytech/polkadot-sdk#1905]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Motivation

We found out that there is currently no reliable method to correlate the initiating extrinsic e.g. `xcmpallet.teleportAssets` with the corresponding DMP message, since the message hash or the message ID is not emitted in an event related to the extrinsic.

Presently, we are employing some heuristics to match messages at origin and destination:

1.  The sender initiates a DMP message through an extrinsic of the `xcmpallet`. We monitor these extrinsics and parse the parameters to obtain the destination paraId
2.  An event `xcmpallet.Attempted` is emitted, indicating the outcome.
3.  If the outcome is `Completed`, the message enters the `dmp.downwardMessageQueues`.
4.  We query the `dmp.downwardMessageQueues` storage at the block of the originating extrinsic, using the destination paraId, to get all messages in the queue.
5.  We parse each message data to extract the instruction set and attempt to match the parameters of the extrinsic with the instructions.
6.  If a match is found, we hash the message data to obtain the message hash and store it for matching at the destination.
7.  When the message is routed to the destination, an event `dmpqueue.ExecutedDownward` is emitted, including the message hash, ID, and outcome. We match the emitted message hash with the one stored in step 6.

These heuristics work for most of the cases, especially considering the low volume of cross-chain messages right now.
However, in the case of multiple DMP messages at the same block for the same destination chain with identical instructions and parameters, it becomes impossible to know, with the heuristics, from which extrinsic they originate.

### Request

We propose emitting an event from the originating extrinsic containing both the message hash and ID. This enhancement will greatly simplify the process of matching messages while also solving the issue of correlating to the originating extrinsic.

### Solution

Found this open PR that should solve the issue in the archived Polkadot repo but I couldn't find it in this repo: [Introduced events for dmp DownwardMessageSent + refactor to common MessageId for events (ump, dmp)](https://github.com/paritytech/polkadot/pull/7161)

Related work done for UMP: [Add event for showing the hash of a UMP sent message](https://github.com/paritytech/cumulus/pull/1228)

### Are you willing to help with this request?

Maybe (please elaborate above)

**Comment by bkontur** [comment]
@XY-Wang yes, that's my [PR](https://github.com/paritytech/polkadot/pull/7161/files) and I have not ported it yet, but I could do it now when needed :)
